integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -295,6 +296,29 @@ def _convert_query_result_to_documents(self, query_result: Dict[str, Any]) -> Li
 
         return documents
 
+    @staticmethod
+    def check_metadata(document: Document):
```

Comment:
```suggestion
    def _validate_metadata(document: Document):
```

I think this should be an internal method. I'm totally ok with a name that expresses better what this method does (`_discard_invalid_meta`).
Please also add a short docstring.